src/backend/distributed/connection/connection_management.c

```diff
@@ -1043,9 +1050,6 @@ StartConnectionEstablishment(ConnectionHashKey *key)
 static void
 FreeConnParamsHashEntryFields(ConnParamsHashEntry *entry)
 {
-	char **keyword = NULL;
```

Comment:
We should probably also reset the size in this function. It's not really necessary anymore after the other fixes, but it seems like a sane thing to do given that we do set keywords and values to NULL.